src/modules/Dropdown/Dropdown.js

```diff
@@ -82,121 +66,611 @@ export default class Dropdown extends Component {
+  }
+
+  handleSearchChange = (e) => {
+    // console.debug('Dropdown.handleSearchChange()')
+    // prevent propagating to handleChange()
+    e.stopPropagation()
+    const newValue = e.target.value
+
+    // console.log({ newValue })
+    this.setState({
+      selectedIndex: 0,
+      searchQuery: newValue,
+    })
+  }
+
+  // ----------------------------------------
+  // Getters
+  // ----------------------------------------
+
+  getOptions = () => {
+    const { options, search } = this.props
+    const { searchQuery } = this.state
+
+    if (!search || !searchQuery) return options
+
+    const re = new RegExp(searchQuery, 'i')
+    return _.filter(options, (opt) => re.test(opt.text))
+  }
+
+  getItemByValue = (value) => {
+    // console.debug('Dropdown.getItemByValue()')
+    return _.find(this.props.options, { value })
+  }
+
+  getSelectedItem = () => {
+    // console.debug('Dropdown.getSelectedItem()')
+    const { selectedIndex } = this.state
+    const options = this.getOptions()
+
+    const selectedItem = options[selectedIndex]
+    // console.log({ selectedIndex, selectedItem, options })
+    return selectedItem
+  }
+
+  getSelectedIndexByValue = (value) => {
+    // console.debug('Dropdown.getSelectedIndexByValue()')
+    const options = this.getOptions()
+    const index = _.findIndex(options, { value })
+    // console.log({ value, options, index })
+    return index
+  }
+
+  // ----------------------------------------
+  // Setters
+  // ----------------------------------------
+
+  setItem = ({ text, value }) => {
+    const { multiple, options } = this.props
+    // console.debug('Dropdown.setItem()')
+    // console.log(item)
+
+    this.onChange(value)
+    this.trySetState({
+      text,
+      value,
+    }, {
+      searchQuery: '',
+      selectedIndex: _.findIndex(options, { value }),
+    })
+
+    if (!multiple) this.close()
+  }
+
+  setItemByValue = (value) => {
+    // console.debug('Dropdown.setItemByValue()')
+    // console.log(value)
+    const item = this.getItemByValue(value)  // { text, value }
+    this.setItem(item)
+  }
+
+  moveSelectionBy = (offset) => {
+    // console.debug('Dropdown.moveSelectionBy()')
+    // console.log(`offset: ${offset}`)
+    const { selectedIndex } = this.state
+
+    const options = this.getOptions()
+    const lastIndex = options.length - 1
+
+    // next is after last, wrap to beginning
+    // next is before first, wrap to end
+    let nextIndex = selectedIndex + offset
+    if (nextIndex > lastIndex) nextIndex = 0
+    else if (nextIndex < 0) nextIndex = lastIndex
+
+    this.setState({ selectedIndex: nextIndex })
+    this.scrollSelectedItemIntoView()
+  }
+
+  // ----------------------------------------
+  // Behavior
+  // ----------------------------------------
+
+  scrollSelectedItemIntoView = () => {
+    const menu = document.querySelector('.ui.dropdown .menu.visible')
+    const item = document.querySelector('.ui.dropdown .menu.visible .item.selected')
```

Comment:
This won't work with multiple dropdowns on the page, consider using refs instead.